ikarus/utils/pathfollowingfunctions.hh

```diff
@@ -41,16 +46,38 @@ namespace Ikarus {
 
     template <typename NonLinearOperator>
     void initialPrediction(NonLinearOperator& nonLinearOperator, SubsidiaryArgs& args) {
-      auto linearSolver = Ikarus::LinearSolver(Ikarus::SolverTypeTag::d_LDLT);  // for the linear predictor step
+      Ikarus::SolverTypeTag solverTag;
+      if constexpr (Ikarus::Std::isSpecializationTypeAndNonTypes<
+                        Eigen::Matrix, std::remove_cvref_t<typename NonLinearOperator::DerivativeType>>::value)
+        solverTag = Ikarus::SolverTypeTag::d_LDLT;
+      else if constexpr (Ikarus::Std::isSpecializationTypeNonTypeAndType<
+                             Eigen::SparseMatrix,
+                             std::remove_cvref_t<typename NonLinearOperator::DerivativeType>>::value)
+        solverTag = Ikarus::SolverTypeTag::sd_SimplicialLDLT;
+      else
+        DUNE_THROW(Dune::NotImplemented,
+                   "Linear solver not implemented for the chosen derivative type of the non-linear operator");
 
       nonLinearOperator.lastParameter() = 1.0;  // lambda =1.0
 
       nonLinearOperator.template update<0>();
       const auto& R = nonLinearOperator.value();
       const auto& K = nonLinearOperator.derivative();
 
-      linearSolver.factorize(K);
-      linearSolver.solve(args.DD, -R);
+      static constexpr bool isLinearSolver
+          = Ikarus::Concepts::LinearSolverCheck<decltype(Ikarus::LinearSolver(solverTag)),
+                                                typename NonLinearOperator::DerivativeType,
+                                                typename NonLinearOperator::ValueType>;
+
+      if constexpr (isLinearSolver) {
+        auto linearSolver = Ikarus::LinearSolver(solverTag);  // for the linear predictor step
+        linearSolver.analyzePattern(K);
+        linearSolver.factorize(K);
+        linearSolver.solve(args.DD, -R);
+      } else {
+        auto linearSolver = Ikarus::SolverDefault{};
+        args.DD           = linearSolver(-R, K);
```

Comment:
I think you will not end up here with a single degree of freedom. Thus, you can static_assert that `isLinearSolver` is true and throw this branch here out